src/main/java/org/elasticsearch/script/ScriptModes.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import org.elasticsearch.ElasticsearchIllegalArgumentException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.script.ScriptService.ScriptType;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+public class ScriptModes {
+
+    static final String SCRIPT_SETTINGS_PREFIX = "script.";
+    static final String ENGINE_SETTINGS_PREFIX = "script.engine";
+
+    private final ESLogger logger;
+
+    final ImmutableMap<String, ScriptMode> scriptModes;
+
+    ScriptModes(Map<String, ScriptEngineService> scriptEngines, Settings settings) {
+        this.logger = Loggers.getLogger(getClass(), settings);
```

Comment:
I think we should rather pass the logger to it or don't log here at all. we only use it for this:

```Java
 logger.warn("ignoring [disable_dynamic] setting value as conflicting scripting settings have been specified");
```

IMO we should rather collect the conflicting settings and provide it as a list and then log in the `ScriptService` with all the settings that are conflicting...